src/pages/admin/index.tsx

```diff
@@ -0,0 +1,413 @@
+} = programs
+
+const SUBDOMAIN = process.env.MARKETPLACE_SUBDOMAIN
+
+export async function getServerSideProps({ req }: NextPageContext) {
+  const subdomain = req?.headers['x-holaplex-subdomain']
+
+  const {
+    data: { marketplace },
+  } = await client.query<GetMarketplace>({
+    query: gql`
+      query GetMarketplace($subdomain: String!) {
+        marketplace(subdomain: $subdomain) {
+          subdomain
+          name
+          description
+          logoUrl
+          bannerUrl
+          ownerAddress
+          creators {
+            creatorAddress
+            storeConfigAddress
+          }
+          auctionHouse {
+            address
+            treasuryMint
+            auctionHouseTreasury
+            treasuryWithdrawalDestination
+            feeWithdrawalDestination
+            authority
+            creator
+            auctionHouseFeeAccount
+            bump
+            treasuryBump
+            feePayerBump
+            sellerFeeBasisPoints
+            requiresSignOff
+            canChangeSalePrice
+          }
+        }
+      }
+    `,
+    variables: {
+      subdomain: subdomain || SUBDOMAIN,
+    },
+  })
+
+  if (isNil(marketplace)) {
+    return {
+      notFound: true,
+    }
+  }
+
+  return {
+    props: {
+      marketplace,
+    },
+  }
+}
+
+interface GetMarketplace {
+  marketplace: Marketplace | null
+}
+
+interface EditPageProps extends AppProps {
+  marketplace: Marketplace
+}
+
+interface NftFilterForm {
+  attributes: AttributeFilter[]
+  preset: PresetEditFilter
+}
+
+const AdminPage: NextPage<EditPageProps> = ({ marketplace }) => {
+  const { connection } = useConnection()
+  const solana = useWallet()
+  const { publicKey } = solana
+  const router = useRouter()
+  const [showEditMarketplace, setShowEditMarketplace] = useState(false)
+  const [logo, setLogo] = useState<string>(marketplace.logoUrl)
+  const [banner, setBanner] = useState<string>(marketplace.bannerUrl)
+
+  // console.log('Marketplace data', marketplace)
+
+  const refreshProps = () => {
+    router.replace(router.asPath)
+  }
+
+  const { watch, register, control } = useForm<NftFilterForm>({
+    defaultValues: { preset: PresetEditFilter.Marketplace },
+  })
+
+  const [preset, setPreset] = useState<PresetEditFilter | undefined>(
+    PresetEditFilter.Marketplace
+  )
+
+  const onAddCreatorClicked = async (form: AddCreatorForm) => {
+    toast('Saving changes...')
+    const { walletAddress } = form
+
+    const creators = [{ address: walletAddress }]
+    marketplace.creators.forEach((creator) => {
+      creators.push({ address: creator.creatorAddress })
+    })
+
+    const inputData = await getInputData(
+      {
+        name: marketplace.name,
+        description: marketplace.description,
+        logo: { url: marketplace.logoUrl },
+        banner: { url: marketplace.bannerUrl },
+        creators,
+      },
+      marketplace,
+      solana
+    )
+
+    updateMarketplace(inputData)
+  }
```

Comment:
Can we use use-react-form for all the state management. We should set default values of the form based on response markeplace API response and try not to manage input changes through it and now with react state.

Reply:
@kespinola By 'use-react-form' do you mean useForm from react-hook-form?

Reply:
yep. instead of having assemble the values in the onSubmit you can use defaultValues to configure the form object based on the current state of marketplace

https://react-hook-form.com/api/useform

Reply:
@kespinola I have made some structural changes in the new commit, please have a look. Also, for the Creators (update) component when a new creator is added, to keep it simple, I am not using react-hook-form. When admin presses 'enter' button (after entering a new creator address), just the new address is being sent to the callback function. That is why in the callback function I am using marketplace api data (but not changing the data, just using it to create new store json)

Similar thing is being done with remove creator as setting all the maketplace data in the defaultValues in this case will be of no use, as we are just tracking remove or add new creator in this Creator component.